### DISCUSSION TITLE
`any()` function ignoring the first character

### DISCUSSION THREAD

**Discussion by Vaimer9** [zesterer/chumsky#114] [Q&A] [answered]
The following code doesn't accept the first character in the provided input:
```rs
let raw = any()
            .ignore_then(text::ident().padded().repeated())
            .padded()
            .collect::<String>()
            .map(Tokens::Raw);
```
So if I were to give an input of `hello world` I would get `elloworld` (I also have padding issues), can someone guide me in the correct direction to fix both the `any()` issue and the padding bug

**Comment by wackbyte** [comment, marked as answer]
- The first character is ignored by `any().ignore_then(...)`, if you don't want to ignore it, use `then`
- `text::ident().padded().repeated()` outputs a list of strings and the `padded()` ignores the whitespace in between. The `collect::<String>()` then merges them all into a single string, so: `"ello world" -> ["ello", "world"] -> "elloworld"`

**Reply by Vaimer9** [reply]
So the following code results in an error
![image](https://user-images.githubusercontent.com/81609819/161466270-a5308e0b-d757-4afd-b25a-497e11f7252b.png)
Here's the error:
![image](https://user-images.githubusercontent.com/81609819/161466352-4f6b7254-d972-4d36-931b-7283a856a0c2.png)

**Reply by zesterer** [reply]
`a.then(b)` results in an output type that is a tuple of `(A, B)` (where `A` and `B` are the output types of `a` and `b` respectively, of course). In your case, the types of `A` and `B` are `char` (because of the `any`) and `Vec<String>` (because of the `text::ident().padded().repeated()`). A tuple cannot be `collect`-ed into a `String`, as Rust explains.

I'm not entirely sure what output you're expecting, so I can't really help you any further with this. What output do you want the parser to produce?

**Reply by Vaimer9** [reply]
The output I want is:
input: `foo bar xyz`
output: `Raw("foo bar xyz")`

**Reply by zesterer** [reply]
Is there a problem with the following?

```rs
text::ident()
    .padded()
    .repeated()
    .map(|idents| idents.join(" "))
```

**Reply by Vaimer9** [reply]
that works perfectly, don't know why that didn't come into mind 😅
